### DISCUSSION TITLE
Migrate Logging to Scribe?

### DISCUSSION THREAD

**Discussion by darkfrog26** [sbt/sbt#7124] [Ideas]
I see in the most recent release notes there was work done to fix the log4j issues. I was wondering if there might be any benefit to migrating to Scribe to remove any Java-based logging issues going forward. I'm obviously biased since I wrote Scribe, but the performance and lack of any third-party dependencies make it a good candidate in my opinion for a tool like SBT.

**Comment by eed3si9n** [comment]
[sbt 1.8.1](https://eed3si9n.com/sbt-1.8.1) fixes slf4j-api 2.x getting pulled in.

sbt 0.13 and prior used hand-rolled logging code, and in sbt 1.x I introduced log4j 2.x, which I had hoped to speed up some operations, but the result I don't think was straightforward, we ended up implementing [another hand-rolled logging midway sbt 1.x](https://github.com/sbt/sbt/pull/5731), and sbt 2.x will not use log4j.

I will to consider another logging library, but I don't personally want to spend my own time investigating if it's worth it. Some notes I can share:
- ephemeral logger are created for each task, and there might be lots of them
- users expect relatively quick response on the stdout logging. for example, `~testQuick` should complete in small latency.
- in general, log displaying the test result or compilation warnings are on the synchronous path of the UX. all results must be displayed before a command returns control to the main loop.
- debug level logs are written into files. sbt has `last` command that can query for a debug log using task name.

None of these are particularly odd, but all combined makes it an interesting combination.

**Reply by darkfrog26** [reply]
All of these things can be accomplished easily with Scribe as it is now. In fact, Scribe is the only logging framework in Scala that would fulfill these requirements as it's the only one (at least that I know of) that doesn't depend on a Java logger. I would be happy to assist and even add functionality to Scribe to properly support anything you need in SBT. It's by far the fastest logging library on the JVM (https://github.com/outr/scribe/wiki/benchmarks) and the most flexible.
